util/util.go

```diff
@@ -276,3 +277,180 @@ func ResourcePagerDutyParseColonCompoundID(id string) (string, string, error) {
 
 	return parts[0], parts[1], nil
 }
+
+func ValidateTZValueDiagFunc(v interface{}, p cty.Path) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	value := v.(string)
+	valid := false
+
+	if sort.SearchStrings(validTZ, value) < len(validTZ) {
```

Comment:
sort.Search has a weird interface, we thus need to check if `validTZ[index] == value` in order have a valid match.

Example: https://go.dev/play/p/Osx8HcsVL1F
<img width="586" alt="image" src="https://github.com/PagerDuty/terraform-provider-pagerduty/assets/12974475/5a77b1c3-65fe-43d8-bf0e-044834b58271">